test/supavisor/db_handler_test.exs

```diff
@@ -26,8 +26,10 @@ defmodule Supavisor.DbHandlerTest do
   describe "handle_event/4" do
     test "db is avaible" do
       :meck.new(:gen_tcp, [:unstick, :passthrough])
-      :meck.expect(:gen_tcp, :connect, fn _host, _port, _socket_opts -> {:ok, :socket} end)
-      :meck.expect(:gen_tcp, :send, fn _socket, _msg -> :ok end)
+      :meck.new(:inet, [:unstick, :passthrough])
+      :meck.expect(:gen_tcp, :connect, fn _host, _port, _sock_opts -> {:ok, :sock} end)
+      :meck.expect(:gen_tcp, :send, fn _sock, _msg -> :ok end)
+      :meck.expect(:inet, :setopts, fn _sock, _opts -> :ok end)
```

Comment:
this is a separate discussion (apologies, I don't think we had one about it!) but while mocking is pretty convenient here I think it provides less confidence and more fragile (totally coupled to the implementation) test suite. Since it's trivial to spawn tcp sockets on the BEAM, I'd consider doing just that and testing against them.

Reply:
that is a good point, added issue #128